src/Storages/MergeTree/registerStorageMergeTree.cpp

```diff
@@ -687,10 +690,61 @@ static StoragePtr create(const StorageFactory::Arguments & args)
                 args.storage_def->ttl_table->ptr(), metadata.columns, context, metadata.primary_key, allow_suspicious_ttl);
         }
 
+        storage_settings->loadFromQuery(*args.storage_def, context, LoadingStrictnessLevel::ATTACH <= args.mode);
+
+        /// Updates the default storage_settings with settings specified via SETTINGS arg in a query
+        if (args.storage_def->settings)
+        {
+            if (args.mode <= LoadingStrictnessLevel::CREATE)
+                args.getLocalContext()->checkMergeTreeSettingsConstraints(initial_storage_settings, storage_settings->changes());
+            metadata.settings_changes = args.storage_def->settings->ptr();
+        }
+
         if (args.query.columns_list && args.query.columns_list->indices)
-            for (auto & index : args.query.columns_list->indices->children)
+        {
+            for (const auto & index : args.query.columns_list->indices->children)
+            {
                 metadata.secondary_indices.push_back(IndexDescription::getIndexFromAST(index, columns, context));
 
+                auto index_name = index->as<ASTIndexDeclaration>()->name;
+                if (((*storage_settings)[MergeTreeSetting::add_minmax_index_for_numeric_columns]
+                    || (*storage_settings)[MergeTreeSetting::add_minmax_index_for_string_columns])
+                    && index_name.starts_with(IMPLICITLY_ADDED_MINMAX_INDEX_PREFIX))
+                {
+                    throw Exception(ErrorCodes::BAD_ARGUMENTS, "Cannot create table because index {} uses a reserved index name", index_name);
+                }
+            }
+
+            /// Try to add "implicit" min-max indexes on all columns
+            for (const auto & column : metadata.columns)
+            {
+                if ((isNumber(column.type) && (*storage_settings)[MergeTreeSetting::add_minmax_index_for_numeric_columns])
+                    || (isString(column.type) && (*storage_settings)[MergeTreeSetting::add_minmax_index_for_string_columns]))
+                {
+                    bool minmax_index_exists = false;
+
+                    for (const auto & index: metadata.secondary_indices)
+                    {
+                        if (index.column_names.front() == column.name && index.type == "minmax")
+                        {
+                            minmax_index_exists = true;
+                            break;
+                        }
+                    }
+
+                    if (minmax_index_exists)
+                        continue;
+
+                    auto index_type = makeASTFunction("minmax");
+                    auto index_ast = std::make_shared<ASTIndexDeclaration>(
+                            std::make_shared<ASTIdentifier>(column.name), index_type,
+                            IMPLICITLY_ADDED_MINMAX_INDEX_PREFIX + column.name);
+                    index_ast->granularity = ASTIndexDeclaration::DEFAULT_INDEX_GRANULARITY;
```

Comment:
@rschu1ze Updated to set default granularity, please let me know if you have any suggestions on the value

Reply:
`DEFAULT_INDEX_GRANULARITY` is fine, tnx